plugins/inputs/logfile/logfile.go

```diff
@@ -35,14 +37,27 @@ type LogFile struct {
 	configs           map[*FileConfig]map[string]*tailerSrc
 	done              chan struct{}
 	removeTailerSrcCh chan *tailerSrc
+	numUsedFds        semaphore.Semaphore
 	started           bool
 }
 
 func NewLogFile() *LogFile {
+	fileDescriptorLimit, err := fdlimit.CurrentFileDescriptorLimit()
+
+	if err != nil {
+		return &LogFile{
+			configs:           make(map[*FileConfig]map[string]*tailerSrc),
+			done:              make(chan struct{}),
+			removeTailerSrcCh: make(chan *tailerSrc, 100),
+			numUsedFds:        semaphore.NewSemaphore(0),
+		}
+	}
+
```

Comment:
Why do we need to catch the error here? We aren't doing anything with the error and just use a default 0 value. This is equivalent to just removing the whole `if` block because the default value returned when the file descriptor check on Unix fails is zero.